src/ImmersedBoundaries/zstar_immersed_grid.jl

```diff
@@ -0,0 +1,58 @@
+using Oceananigans.Grids: AbstractZStarGrid
+using Oceananigans.Operators
+using Oceananigans.Operators: ZSRG, ZSLLG, ZSOSG, superscript_location
+
+import Oceananigans.Grids: dynamic_column_depthᶜᶜᵃ, 
+                           dynamic_column_depthᶜᶠᵃ,
+                           dynamic_column_depthᶠᶜᵃ,
+                           dynamic_column_depthᶠᶠᵃ
+
+import Oceananigans.Operators: σⁿ, σ⁻, ∂t_σ
+
+const ZStarImmersedGrid   = ImmersedBoundaryGrid{<:Any, <:Any, <:Any, <:Any, <:AbstractZStarGrid}
+const ZStarGridOfSomeKind = Union{ZStarImmersedGrid, AbstractZStarGrid}
+
+@inline dynamic_column_depthᶜᶜᵃ(i, j, k, grid::ZStarGridOfSomeKind, η) = static_column_depthᶜᶜᵃ(i, j, grid) +  @inbounds η[i, j, k]
```

Comment:
```suggestion
@inline column_depthᶜᶜᵃ(i, j, k, grid::ZStarGridOfSomeKind, η) = static_column_depthᶜᶜᵃ(i, j, grid) +  @inbounds η[i, j, k]
```

This is just the "column depth" isn't it? The concept of dynamic vs static doesn't make sense for a fixed grid. Better to refer to this as something that generalizes to all grids. Then we have "static column depth" for the specific case that it's needed.